### PR TITLE
GH-14721: add new method and new button name, adjust method

### LINKED ISSUE #14721: Rename Exit Button when navigate from Cart
Rename Exit Button when navigate from Cart

- Exit button should be named "Cancel Configuration" when navigate from cart 

- The button should then bring the user back to cart when clicked

### PR DESCRIPTION
**Pull request by Changsuwan**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.ts [hunk 1]
```diff
@@ -46,7 +58,15 @@ export class ConfiguratorExitButtonComponent {
         take(1)
       )
       .subscribe((product) =>
-        this.routingService.go({ cxRoute: 'product', params: product })
+        this.routerData$.pipe(take(1)).subscribe((routerData) => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.ts]: Could you please simplify this expression and only subscribe once on this.configRouterExtractorService
      .extractRouterData()?

File: feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.ts [hunk 2]
```diff
@@ -5,28 +5,40 @@ import {
...
 
   constructor(
     protected productService: ProductService,
     protected routingService: RoutingService,
     protected configRouterExtractorService: ConfiguratorRouterExtractorService,
     protected configuratorCommonsService: ConfiguratorCommonsService,
     protected breakpointService: BreakpointService,
-    protected windowRef: WindowRef
+    protected windowRef: WindowRef,
+    protected location: Location
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.ts:61]: Please add an issue that covers this breaking change.  See e.g. #14352

File: feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.html [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
-  (click)="exitConfiguration()"
->
-  <ng-container *ngIf="isDesktop() | async">
-    {{ 'configurator.button.exit' | cxTranslate }}
-  </ng-container>
-  <ng-container *ngIf="isMobile() | async">
-    {{ 'configurator.button.exitMobile' | cxTranslate }}
-  </ng-container>
-</button>
+<ng-container *ngIf="routerData$ | async as routerData">
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.html]: Please describe the changes in 5_0.md

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.ts (hunk 2 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.ts:61]: An issue has been created. Please check if it is correctly done.

File: feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.ts (hunk 1 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.ts]: Done

File: feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.html (hunk 3 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/components/exit-button/configurator-exit-button.component.html]: Done

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**